### PR TITLE
Feature custom rules, simplified #1

### PR DESCRIPTION
**Pull request by abbadon1334**
#1 a simplified version of the previous PR

in place of write :
```php
        $model = $model->newInstance();
        $model->addCondition($field, '=', $value);
        
        $action = $model->action('count');
        $count  = (int)$action->getOne();
                
        // if is loaded must be 1;
        if ($isLoaded && $count == 1) {
            return true;
        }
        
        // if is not loaded must be 0;
        if (!$isLoaded && $count == 0) {
            return true;
        }
```
can be used in validation definition like this :

```php
$validator->rule('email', [ ['atk4_value_unique', $userModel] ]);
```

### CODE REVIEW

**Comment by abbadon1334** [comment]
@romaninsh i think this can be interesting to be integrated, but i have doubts on naming.

**Comment by romaninsh** [comment]
@DarkSide666 was working on "Unique" property for fields, which would be doing exactly same, but it would also be on ATK Data level not in a validator.

There is an error in your logic.

 - client A has email "foo@bar.baz"
 - edit Client B, set his email to "foo@bar.baz" - will allow you to save.

You need this logic:

``` php
 $q-> addCondition('email', $value)
 if($loaded) $q->addCondition('id', '!=', $this->id);
if ($q->action('count') > 0) { fail }
```

File: src/Rules/ValueUniqueOther.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                    'params' => $params,
+                    'fields' => $fields,
+                ]
+            );
+        }
+
+        $isLoaded = $ref_model->loaded(); // if loaded ? INSERT : UPDATE
+
+        $count = (int) $ref_model->newInstance()
+                                ->addCondition($ref_model_field, '=', $value)
```
Comment (romaninsh) [inline src/Rules/ValueUniqueOther.php:65]: can you provide a real-life example (simple one), if you can't come up with a good one - then lets exclude the rule (as it wouldn't be helpful)

File: src/Rules/ValueUniqueOther.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+use atk4\data\Model;
+
+/**
+ * Class ValueUniqueOther.
+ *
+ * Validate : if value is unique in Model in another field
+ *
+ * Rule code : atk4_value_unique_other
+ *
+ * @example ['atk4_value_unique_other',$model,'field_name_to_check']
```
Comment (romaninsh) [inline src/Rules/ValueUniqueOther.php:15]: give full example here with real-life fields: $model->addField('..')->validate(...)

**Review by romaninsh — CHANGES_REQUESTED** [review]
few things to take care of.

**Comment by abbadon1334** [comment]
i will rebuild essentials ones for 3.1 follow suggestion of Romans

**Pull request closed without merging**